cpp/src/strings/utilities.cuh

```diff
@@ -30,20 +30,35 @@ namespace strings
 namespace detail
 {
 
+/**
+ * @brief This utility will copy the argument string's data into
```

Comment:
What a weird utility. Are you sure this 2-line function needs to be a function? It's very weird to return a pointer to the end of a string rather than the beginning, especially when the length is known.

```suggestion
 * @brief Copies input string data into a buffer and increments the pointer by the number of bytes copied.
```